src/lib/sql/random_cache.hpp

```diff
@@ -84,6 +84,11 @@ class RandomCache : public AbstractCache<Key, Value> {
   void _evict() {
     _map.erase(_list[0].first);
     _list.erase(_list.cbegin());
```

Comment:
Maybe an item can be deleted from the back instead of the front so that there is no need for the extra loop. Then this function can continue to run in O(1).

Reply:
We decided to remove elements from the front during resizing because of our query cache demo, but I do not remember the exact reason.

Reply:
Ok, I just wanted to check that there is a reason for this overhead.